rest/index.py

```diff
@@ -468,6 +481,433 @@ def createIncident():
+    ```
+
+    Where:
+        * `"alert"` is the "label" for the alert, signifying which of the
+        supported alerts is being triaged.
+        * `"user"` is the email address of the user contacted.
+        * `"identifier"` is a list of ElasticSearch IDs of alerts of the
+        same kind triggered by the same user.
+
+    This function writes back a response containing the following JSON.
+
+    ```
+    {
+        "error": Optional[str]
+    }
+    ```
+
+    If an error occurs, a duplicate chain will not be created and an error
+    string will be returned.  Otherwise, the `error` field will be `null.`
+    """
+
+    response.content_type = "application/json"
+
+    initConfig()
+
+    mongo = MongoClient(options.mongohost, options.mongoport)
+    dupchains = mongo.meteor[DUP_CHAIN_DB]
+
+    try:
+        req = json.loads(request.body.read())
+        request.body.close()
+    except ValueError:
+        response.status = StatusCode.BAD_REQUEST
+        response.body = json.dumps({"error": "Missing or invalid request body"})
+        return response
+
+    now = datetime.utcnow()
+
+    chain = {
+        "alert": req.get("alert"),
+        "user": req.get("user"),
+        "identifiers": req.get("identifiers", []),
+        "created": now,
+        "modified": now,
+    }
+
+    if chain["alert"] is None or chain["user"] is None:
+        response.status = StatusCode.BAD_REQUEST
+        response.body = json.dumps(
+            {"error": "Request missing required key `alert` or `user`"}
+        )
+        return response
+
+    result = dupchains.insert_one(chain)
+    if not result.acknowledged:
+        response.status = StatusCode.INTERNAL_ERROR
+        response.body = json.dumps({"error": "Failed to store new duplicate chain"})
+        return response
+
+    response.status = StatusCode.OK
+    response.body = json.dumps({"error": None})
+
+    return response
+
+
+@put("/alerttriagechain")
+@put("/alerttriagechain/")
+def update_duplicate_chain():
+    """Update a `DuplicateChain`, appending information about a new alert
+    destined for a Slack user via the triage Bot.
+    See `create_duplicate_chain` for more information.
+
+    Requests are expected to take the following (JSON) form:
+
+    ```
+    {
+        "alert": str,
+        "user": str,
+        "identifiers": List[str]
+    }
+    ```
+
+    The parameters are the same as those of `create_duplicate_chain`.
+
+    This function writes back a response containing the following JSON.
+
+    ```
+    {
+        "error": Optional[str]
+    }
+    ```
+
+    If an error occurs, no duplicate chains will be updated.  This endpoint
+    does not create a new chain if one does not already exist.
+    """
+
+    response.content_type = "application/json"
+
+    initConfig()
+
+    mongo = MongoClient(options.mongohost, options.mongoport)
+    dupchains = mongo.meteor[DUP_CHAIN_DB]
+
+    try:
+        req = json.loads(request.body.read())
+        request.body.close()
```

Comment:
You could use [bottle's native support for POSTed JSON](https://bottlepy.org/docs/dev/api.html#bottle.BaseRequest.json)